package baidu.lumia;

option cc_generic_services = true;
option py_generic_services = true;

message DeviceStatus {
    optional string name = 1;
    optional bool healthy = 2;
}

message MountStatus {
    optional string dev = 1;
    optional string mount_point = 2;
    optional bool mounted = 3;
}

message MinionStatus {
    repeated DeviceStatus devices = 1;
    repeated MountStatus mounts = 2;
    optional int64 datetime = 4;
    optional bool all_is_well = 5;
    optional double cpu_used = 6;
}

message QueryAgentRequest {}

message QueryAgentResponse {
    optional MinionStatus minion_status = 1;
    optional int32 status = 2;
    optional string ip = 3;
}

message InitGalaxyEnvRequest {
    optional bool hybrid = 1;
    optional string data_center = 2;
    optional int32 disk_size = 3;
    optional int32 bandwidth = 4;
}

message InitGalaxyEnvResponse {
    optional string ip = 1;
    optional int32 status = 2;
}

message RemoveGalaxyEnvRequest {
    optional bool hybrid = 1;
}

message RemoveGalaxyEnvResponse {
    optional int32 status = 1;
    optional string ip = 2;
}

message ExecRequest {
    optional string cmd = 1;
    optional string script = 2;
}

message ExecResponse {
    optional string ip = 1;
    optional int32 status = 2;
}

service LumiaAgent {
    rpc Query(QueryAgentRequest) returns (QueryAgentResponse);
    rpc Exec(ExecRequest) returns (ExecResponse);
    rpc InitGalaxyEnv(InitGalaxyEnvRequest) returns (InitGalaxyEnvResponse);
    rpc RemoveGalaxyEnv(RemoveGalaxyEnvRequest) returns (RemoveGalaxyEnvResponse);
}


